src/controllers/expenses.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const {
+  getAllExpenses,
+  addExpense,
+  getExpense,
+  deleteExpense,
+  updateExpense,
+} = require('../services/exprensesService');
+const { getAllUsers } = require('../services/usersServer.js');
+
+async function getAllExpensesAction(req, res) {
+  try {
+    const expenses = await getAllExpenses();
+
+    res.json(expenses);
+  } catch (error) {
+    res.sendStatus(500);
+  }
+};
+
+async function addExpenseAction(req, res) {
+  const expense = req.body;
+  const allowedKeys = [
+    'title',
+    'amount',
+    'category',
+    'note',
+    'userId',
+  ];
+  const keysExpense = Object.keys(expense);
+
+  const missingFields = keysExpense.every(key => allowedKeys.includes(key));
+
+  try {
+    const users = await getAllUsers();
+
+    if (
+      !users.length || !users.filter((el) => el.id === expense.userId).length
+    || !missingFields) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const newExpense = await addExpense(expense);
+
+    res.json(newExpense);
+    res.status(201);
```

Comment:
Move the 'res.status(201)' line before 'res.json(newExpense)' to set the correct status code.